src/number_theory/cyclotomic/basic.lean

```diff
@@ -204,6 +209,104 @@ lemma number_field [h : number_field K] [fintype S] [is_cyclotomic_extension S K
+lemma integral [is_domain B] [is_noetherian_ring A] [fintype S] [is_cyclotomic_extension S A B] :
+  algebra.is_integral A B :=
+is_integral_of_noetherian $ is_noetherian_of_fg_of_noetherian' $ (finite S A B).out
+
+/-- If `S` is finite and `is_cyclotomic_extension S K A`, then `finite_dimensional K A`. -/
+lemma finite_dimensional [fintype S] [algebra K A] [is_domain A] [is_cyclotomic_extension S K A] :
+  finite_dimensional K A :=
+finite S K A
+
 end fintype
 
+section
+
+variables {A B}
+
+lemma adjoin_roots_cyclotomic_eq_adjoin_nth_roots [decidable_eq B] [is_domain B] {ζ : B}
+  (hζ : is_primitive_root ζ n) :
+  adjoin A ↑((map (algebra_map A B) (cyclotomic n A)).roots.to_finset) =
+  adjoin A {b : B | ∃ (a : ℕ+), a ∈ ({n} : set ℕ+) ∧ b ^ (a : ℕ) = 1} :=
+begin
+  simp only [mem_singleton_iff, exists_eq_left, map_cyclotomic],
+  refine le_antisymm (adjoin_mono (λ x hx, _)) (adjoin_le (λ x hx, _)),
+  { simp only [multiset.mem_to_finset, finset.mem_coe,
+               map_cyclotomic, mem_roots (cyclotomic_ne_zero n B)] at hx,
+    simp only [mem_singleton_iff, exists_eq_left, mem_set_of_eq],
+    rw is_root_of_unity_iff n.pos,
+    exact ⟨n, nat.mem_divisors_self n n.ne_zero, hx⟩ },
+  { simp only [mem_singleton_iff, exists_eq_left, mem_set_of_eq] at hx,
+    obtain ⟨i, hin, rfl⟩ := hζ.eq_pow_of_pow_eq_one hx n.pos,
+    refine set_like.mem_coe.2 (subalgebra.pow_mem _ (subset_adjoin _) _),
+    rwa [finset.mem_coe, multiset.mem_to_finset, mem_roots $ cyclotomic_ne_zero n B],
+    exact is_root_cyclotomic n.pos hζ }
+end
+
+end
+
+section field
+
+variable [ne_zero ((n : ℕ) : K)]
+
+/-- If `ne_zero ((n : ℕ) : K)`, a cyclotomic extension splits `X ^ n - 1` if `n ∈ S`.-/
+lemma splits_X_pow_sub_one [H : is_cyclotomic_extension S K L] (hS : n ∈ S) :
+  splits (algebra_map K L) (X ^ (n : ℕ) - 1) :=
+begin
+  rw [← splits_id_iff_splits, polynomial.map_sub, polynomial.map_one,
+      polynomial.map_pow, polynomial.map_X],
+  obtain ⟨z, hz⟩ := ((is_cyclotomic_extension_iff _ _ _).1 H).1 hS,
+  rw [aeval_def, eval₂_eq_eval_map, map_cyclotomic] at hz,
+  haveI : ne_zero ((n : ℕ) : L) := ne_zero.of_no_zero_smul_divisors K L,
+  exact X_pow_sub_one_splits (is_root_cyclotomic_iff.1 hz),
+end
+
+/-- If `ne_zero ((n : ℕ) : K)`, a cyclotomic extension splits `cyclotomic n K` if `n ∈ S`.-/
+lemma splits_cyclotomic [is_cyclotomic_extension S K L] (hS : n ∈ S) :
+  splits (algebra_map K L) (cyclotomic n K) :=
+begin
+  refine splits_of_splits_of_dvd _ (X_pow_sub_C_ne_zero n.pos _)
+    (splits_X_pow_sub_one n S K L hS) _,
+  use (∏ (i : ℕ) in (n : ℕ).proper_divisors, polynomial.cyclotomic i K),
+  rw [(eq_cyclotomic_iff n.pos _).1 rfl, ring_hom.map_one],
+end
+
+section singleton
+
+variables [is_cyclotomic_extension {n} K L]
+
+/-- If `is_cyclotomic_extension {n} K L` and `ne_zero ((n : ℕ) : K)`, then `L` is the splitting
+field of `X ^ n - 1`. -/
+lemma splitting_field_X_pow_sub_one : is_splitting_field K L (X ^ (n : ℕ) - 1) :=
```

Comment:
Idem dito, I feel like this should be a one-line consequence of the next lemma. Does that make sense?

Reply:
In general this is false in both direction (if `P ∣Q` there is no relation between being the splitting field of `P` and of `Q`, the extension can be too small to split `Q` and too big to be the splitting field of `P`).

Reply:
Hmm, but in this case a field splits `cyclotomic n` if and only if it splits `X^n - 1`. I was hoping that we could somehow use that.

Reply:
Yeah, but this is because the roots of `X^n-1` are powers of the roots of `cyclotomic n K`, this seems very specific and difficult to abstract.